tests/zfs-tests/tests/functional/zvol/zvol_misc/zvol_misc_snapdev.ksh

```diff
@@ -148,6 +148,12 @@ log_must zfs set snapdev=hidden $ZVOL
 check_missing $SNAPDEV
 log_must zfs set snapdev=hidden $ZVOL
 check_missing $SNAPDEV
+# 2.4 Verify we don't hit a race condition setting snapdev in rapid succession
```

Comment:
That in unfortunate, I think the critical thing to test here is the original `zv_open_count` race this patch addresses.  The good news is that can be largely testing by not hitting the ASSERT. 

There's not much we can do on the udev side of things if the concurrent event handling is the problem.  That said, this is something of a contrived real world use case so I think it would be ok to simply do the following after the loop.

```
block_device_wait
log_must zfs set snapdev=hidden $ZVOL
check_missing $SNAPDEV
```